components/pkg-export-kubernetes/src/manifest.rs

```diff
@@ -151,13 +153,19 @@ impl Manifest {
                       environment })
     }
 
-    /// Generates the manifest as a string and writes it to `write`.
+    /// Generates manifest as a string and writes it to `write`.
     pub fn generate(&mut self, write: &mut dyn Write) -> Result<()> {
         let out: String = ManifestJson::new(&self).into();
 
         write.write_all(out.as_bytes())?;
         Ok(())
     }
+
+    // Replaces any periods in the metadata name with hyphens
+    // To make it a valid Kubernetes resource name
+    fn formatted_metadata_name(metadata_name: &str) -> String {
```

Comment:
Since this function is really just wrapping the replace function I think it and the accompanying test can be removed

Reply:
For thoroughness sake (and to provide a good examples for future refactoring) I would much rather leave it as a separate method that can easily be tested. 

The "new_from_cli_matches" method which calls this method is extremely difficult to test (it currently has no test coverage on it - the existing tests cover the "generate" method, but not the "new_from _cli_matches" method). This exporter does need a much larger refactor to make it easier to test and to work in, but I chose the simpler, easier to test solution for now.

Reply:
I disagree... this is surfacing the heretofore unhandled and unacknowledged fact that we need to be generating names in a particular way for them to be valid. Breaking it out as a separate function with tests is an important first step in the future refactoring that @nellshamrell mentioned.

Reply:
I agree with the separate function for `Manifest::formatted_metadata_name` since it conveys *why* we're doing this rather than just _what_ we're doing as inlining the `replace` call would.

As far as the test goes, I think we can do without it for now since it's simple enough to verify by inspection that I don't think it's worth the extra code/CI time. But if this function were to become more complex, I'd say that's an ideal time to add a test which serves to better document it and avoid regression. That's another benefit of factoring this out into it's own function; it's less work to add the test later and therefore more likely to happen.

Reply:
Hey @baumanj - thank you for the feedback. I would prefer to leave the test in, personally, to avoid potentially regressions as the giant "new_from_cli_matches" method is refactored in the future. Ideally, formatted_metadata_name would be tested through an integration test on the larger "new_from_cli_matches" method. When that integration test is eventually added, we will no longer have need for this unit test. While we are in this middle ground - particularly as this is a tricky bit of the code base and needs to be so specific to be valid for Kubernetes - I would prefer to leave the unit test in. I feel the code time has already been spent and the CI time is minimal at this point.

Reply:
That sounds good. We can have differences of opinion, and it's not worth spending too much energy on this particular point. I like when we all feel comfortable sharing our thoughts constructively and then defer to the author's prerogative so that we can keep things moving. When I feel strongly about something as a reviewer, I try to say why and find a common understanding. The existence of this one test isn't something I feel strongly about, but I appreciate you hearing my feedback.